dvc/repo/pull.py

```diff
@@ -28,6 +29,7 @@ def pull(
         all_commits=all_commits,
         with_deps=with_deps,
         recursive=recursive,
+        drop_index=drop_index,
```

Comment:
should we start using `**kwargs` @pmrowla @efiop ? it seems we have to change a lot of method just to pass one arguments, and it's not very scalable - not ideal to have functions with 9 args?

Reply:
I think using `**kwargs` would make sense, but we'll want to make sure the expected params are documented properly. I think a lot of the top level function definitions for push/pull/gc/etc are currently missing docstrings.

Reply:
created separate issue for this: #3612